lib/internal/bootstrap/loaders.js

```diff
@@ -208,6 +210,74 @@
         NativeModule.require;
       fn(this.exports, requireFn, this, process);
 
+      if (config.experimentalModules) {
+        if (this.id === 'events') {
+          this.namespace = [];
+        } else {
+          this.namespace = Object.entries(
+            Object.getOwnPropertyDescriptors(this.exports))
+            .filter(([name, d]) => d.enumerable)
+            .map(([name]) => name);
+        }
+
+        const update = (property, value) => {
+          if (this.reflect !== undefined && this.namespace.includes(property))
+            this.reflect.exports[property].set(value);
+        };
+
+        const methodWrapMap = new WeakMap();
+
+        const proxy = new Proxy(this.exports, {
+          set(target, prop, value) {
```

Comment:
`set` hook has a fourth `receiver` argument that you are not handling.
You need to at least make sure to forward the `receiver` argument to `Reflect.set`.